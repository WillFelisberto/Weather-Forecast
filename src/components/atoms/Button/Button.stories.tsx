import type { Meta, StoryObj } from '@storybook/react';

import { Button } from './Button';

const meta: Meta<typeof Button> = {
  title: 'Atoms/Button',
  component: Button,
  tags: ['autodocs'],
  args: {
    children: 'Example'
  },
  argTypes: {
    variant: {
      options: ['default', 'primary', 'success', 'warning', 'danger'],
      control: { type: 'select' }
    }
  }
};

export default meta;

type Story = StoryObj<typeof Button>;

export const Default: Story = {};
export const Primary: Story = {
  args: {
    variant: 'primary'
  }
};

export const Success: Story = {
  args: {
    variant: 'success'
  }
};

export const Warning: Story = {
  args: {
    variant: 'warning'
  }
};

export const Danger: Story = {
  args: {
    variant: 'danger'
  }
};
