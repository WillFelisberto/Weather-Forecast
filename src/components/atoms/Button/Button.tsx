import clsx from 'clsx';
import { ButtonHTMLAttributes } from 'react';

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  variant?: 'default' | 'primary' | 'success' | 'warning' | 'danger';
}

export const Button: React.FC<ButtonProps> = ({
  children,
  variant = 'default',
  className,
  ...props
}) => {
  return (
    <button
      className={clsx(
        'px-4 py-2 rounded-xl text-white font-medium transition-all',
        {
          'bg-gradient-to-r from-[#3A5E72] to-[#5C93B1] hover:opacity-80': variant === 'default',
          'bg-gradient-to-r from-[#4B7A90] to-[#6AA8C8] hover:opacity-80': variant === 'primary',
          'bg-gradient-to-r from-[#2D8659] to-[#4BBF87] hover:opacity-80': variant === 'success',
          'bg-gradient-to-r from-[#D0A24B] to-[#F2C977] hover:opacity-80': variant === 'warning',
          'bg-gradient-to-r from-[#A54242] to-[#E66464] hover:opacity-80': variant === 'danger'
        },
        className
      )}
      {...props}
    >
      {children}
    </button>
  );
};
