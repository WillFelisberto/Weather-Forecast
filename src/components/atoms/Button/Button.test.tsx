import { render } from '@/tests/test-utils';

import { Button } from './Button';

// Snapshot Test Suite

describe('Button Component', () => {
  it('renders the default button correctly', () => {
    const { asFragment } = render(<Button>Default Button</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the primary button correctly', () => {
    const { asFragment } = render(<Button variant="primary">Primary Button</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the success button correctly', () => {
    const { asFragment } = render(<Button variant="success">Success Button</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the warning button correctly', () => {
    const { asFragment } = render(<Button variant="warning">Warning Button</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the danger button correctly', () => {
    const { asFragment } = render(<Button variant="danger">Danger Button</Button>);
    expect(asFragment()).toMatchSnapshot();
  });
});
