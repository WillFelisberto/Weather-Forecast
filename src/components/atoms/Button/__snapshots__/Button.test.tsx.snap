// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders the danger button correctly 1`] = `
<DocumentFragment>
  <button
    class="px-4 py-2 rounded-xl text-white font-medium transition-all bg-gradient-to-r from-[#A54242] to-[#E66464] hover:opacity-80"
  >
    Danger Button
  </button>
</DocumentFragment>
`;

exports[`Button Component renders the default button correctly 1`] = `
<DocumentFragment>
  <button
    class="px-4 py-2 rounded-xl text-white font-medium transition-all bg-gradient-to-r from-[#3A5E72] to-[#5C93B1] hover:opacity-80"
  >
    Default Button
  </button>
</DocumentFragment>
`;

exports[`Button Component renders the primary button correctly 1`] = `
<DocumentFragment>
  <button
    class="px-4 py-2 rounded-xl text-white font-medium transition-all bg-gradient-to-r from-[#4B7A90] to-[#6AA8C8] hover:opacity-80"
  >
    Primary Button
  </button>
</DocumentFragment>
`;

exports[`Button Component renders the success button correctly 1`] = `
<DocumentFragment>
  <button
    class="px-4 py-2 rounded-xl text-white font-medium transition-all bg-gradient-to-r from-[#2D8659] to-[#4BBF87] hover:opacity-80"
  >
    Success Button
  </button>
</DocumentFragment>
`;

exports[`Button Component renders the warning button correctly 1`] = `
<DocumentFragment>
  <button
    class="px-4 py-2 rounded-xl text-white font-medium transition-all bg-gradient-to-r from-[#D0A24B] to-[#F2C977] hover:opacity-80"
  >
    Warning Button
  </button>
</DocumentFragment>
`;
